### PR TITLE
 docs: Add Wireguard Getting Started Guide 

### PR DESCRIPTION
**Pull request by gandro**
This splits the `Transparent Encryption` chapter into a IPsec and
Wireguard part and adds a basic getting started guide for Wireguard.

*Best reviewed per commit to avoid confusion due to file renaming.*

**Netlify Preview**: https://nervous-ptolemy-1de0e7.netlify.app/

### CODE REVIEW

**Comment by gandro** [comment]
test-docs-please

File: Documentation/gettingstarted/encryption-wireguard.rst
Comment (gandro) [inline Documentation/gettingstarted/encryption-wireguard.rst:46]: Reviewer note: This is functional as is, but might changed in https://github.com/cilium/cilium-cli/pull/177

File: Documentation/gettingstarted/encryption-wireguard.rst
Comment (gandro) [inline Documentation/gettingstarted/encryption-wireguard.rst:58]: Reviewer note: This is functional as is, but might changed in https://github.com/cilium/cilium/pull/15628

File: Documentation/gettingstarted/encryption-wireguard.rst
Comment (kkourt) [inline Documentation/gettingstarted/encryption-wireguard.rst]:
```suggestion
    Wireguard support in Cilium currently lacks the following features,
```

File: Documentation/gettingstarted/encryption-wireguard.rst
Comment (kkourt) [inline Documentation/gettingstarted/encryption-wireguard.rst]:
```suggestion
    - Host-level encryption. Only traffic between two Cilium-managed endpoints (i.e. pod to pod traffic)
```

File: Documentation/gettingstarted/encryption-wireguard.rst
Comment (kkourt) [inline Documentation/gettingstarted/encryption-wireguard.rst]:
```suggestion
.. code-block:: shell-session
```

see: https://github.com/cilium/cilium/issues/4143

File: Documentation/gettingstarted/encryption-wireguard.rst
Comment (kkourt) [inline Documentation/gettingstarted/encryption-wireguard.rst]:
```suggestion
.. code-block:: shell-session
```
see: https://github.com/cilium/cilium/issues/4143

File: Documentation/gettingstarted/encryption.rst
Comment (kkourt) [inline Documentation/gettingstarted/encryption.rst]: If I'm not mistaken, IPSec also supports host-level encryption. If that's indeed the case, it might be worth mentioning here.

**Review by kkourt — APPROVED** [review]
Looks good to me. Some suggestions for your consideration.

File: Documentation/gettingstarted/encryption-wireguard.rst
Comment (brb) [inline Documentation/gettingstarted/encryption-wireguard.rst]: I think in the upcoming architecture section we should extend this saying that the priv key is stored in /var/run/cilium and it does not survive node restarts.

File: Documentation/gettingstarted/encryption-wireguard.rst
Comment (brb) [inline Documentation/gettingstarted/encryption-wireguard.rst:86]: IMHO, let's get rid of this `Install tcpdump`. I assume that if a user wants to enable wireguard, they will have enough knowledge how to install tcpdump. Also, it's Ubuntu / Debian specific, and not all users run those distros.

**Review by brb — COMMENTED** [review]
LGTM, wg :rocket: :moon: ! Non-blocking comments.

File: Documentation/gettingstarted/encryption-wireguard.rst
Comment (gandro) [reply Documentation/gettingstarted/encryption-wireguard.rst:86]: This is inside the cilium pod, where we actually are using a Debian base image. For managed k8s, the user might not even have access to the node.

**Comment by gandro** [comment]
Thanks for the feedback @brb and @kkourt, I will address it!

File: Documentation/gettingstarted/encryption-wireguard.rst
Comment (brb) [reply Documentation/gettingstarted/encryption-wireguard.rst:86]: Aha! Then makes sense.

**Review by qmonnet — APPROVED** [review]

File: Documentation/gettingstarted/encryption-ipsec.rst
Comment (qmonnet) [inline Documentation/gettingstarted/encryption-ipsec.rst:57]: Before we merge, can you please check that this block is rendered correctly? I know you didn't touch it other than the renaming, but I wonder if the lack of empty line between directive name and block content might result in some error. Might be a good occasion to fix if that were the case.

Comment (gandro) [reply Documentation/gettingstarted/encryption-ipsec.rst:57]: Thanks for the review! It looks like it's rendered correctly to me (from the preview link in PR description): https://nervous-ptolemy-1de0e7.netlify.app/gettingstarted/encryption-ipsec.html#generate-import-the-psk

Comment (qmonnet) [reply Documentation/gettingstarted/encryption-ipsec.rst:57]: Ok my bad then, thanks for checking!

**Comment by pchaigno** [comment]
Isn't the WireGuard integration in beta? We usually indicate that in the doc page title and add a note at the beginning (see e.g. https://nervous-ptolemy-1de0e7.netlify.app/gettingstarted/alibabacloud-eni.html).

**Comment by gandro** [comment]
> Isn't the WireGuard integration in beta? We usually indicate that in the doc page title and add a note at the beginning (see e.g. https://nervous-ptolemy-1de0e7.netlify.app/gettingstarted/alibabacloud-eni.html).

My understanding of labeling something as beta usually means "we're not sure yet about the design and if we want to support this". I do not think this is the case with Wireguard.

While the current implementation of Wireguard has clearly documented limitations, I believe the design is solid and we know this is a feature we willing to support. We already know of users who are testing this feature in staging environments right now and they will likely use it in production environments with 1.10 because for them the limitations are not blocking. So a I would opt for not marking this as beta.

**Comment by pchaigno** [comment]
> My understanding of labeling something as beta usually means "we're not sure yet about the design and if we want to support this". I do not think this is the case with Wireguard.

That's not my understanding. I understand "beta" as meaning the feature hasn't gone through the same extensive testing as other features. For example, the host firewall, IPSec, and deny policies were in beta even though AFAIK we don't plan to change their design. We considered calling that tech preview at some point; maybe that was a better name. My understanding from past discussions was that all new features should be introduced as "beta".

On the specific case of WireGuard, has this question been raised and discussed somewhere? If not, might be good to raise it just to check everyone's on the same page. I'll merge this in the meantime; I don't think we need to block on it.

**Pull request merged**
